# HG changeset patch
# User sbr <sebastien@brissaud.name>
# Date 1368872476 0
# Branch fs5503_02
# Node ID 64eea9358be0daabfec78ff2b1319edf7274456a
# Parent  d837e223c99d7c6e83c87b8ba6708394058a458f
# Parent  af545f6533d43bc507d2d388116582b9f36c903e
When opening the BuildObjectWindow, ensure the selected object is visible in the object selector.

diff --git a/src/object_gui.cpp b/src/object_gui.cpp
--- a/src/object_gui.cpp
+++ b/src/object_gui.cpp
@@ -85,6 +85,7 @@
 		NWidgetMatrix *matrix = this->GetWidget<NWidgetMatrix>(WID_BO_SELECT_MATRIX);
 		matrix->SetScrollbar(this->GetScrollbar(WID_BO_SELECT_SCROLL));
 		matrix->SetCount(ObjectClass::Get(_selected_object_class)->GetUISpecCount());
+		if (_selected_object_index != -1) matrix->SetClicked(_selected_object_index);
 	}
 
 	virtual ~BuildObjectWindow()
